src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -4563,26 +4563,25 @@ def create_image_version(cmd, resource_group_name, gallery_name, gallery_image_n
                 data_snapshots[i] = resource_id(
                     subscription=get_subscription_id(cmd.cli_ctx), resource_group=resource_group_name,
                     namespace='Microsoft.Compute', type='snapshots', name=s)
-    source = GalleryArtifactSource(managed_image=ManagedArtifact(id=managed_image))
-    profile = ImageVersionPublishingProfile(exclude_from_latest=exclude_from_latest,
-                                            end_of_life_date=end_of_life_date,
-                                            target_regions=target_regions or [TargetRegion(name=location)],
-                                            source=source, replica_count=replica_count,
-                                            storage_account_type=storage_account_type)
-    if replication_mode is not None:
-        profile.replication_mode = replication_mode
```

Comment:
Is the reason for splitting this logic because `source` is not supported when api-version >= `2019-07-01`?
If so, could we keep the original logic, just give the value of `source` to `profile` when api-version is less than `2019-07-01`?

Reply:
Yes, the `source` is not supported in `GalleryImageVersionPublishingProfile` when api-version >= 2019-07-01. I will adjust it.